src/main/java/ch/njol/skript/effects/EffTeleport.java

```diff
@@ -125,14 +128,26 @@ protected TriggerItem walk(Event e) {
 			return next;
 		}
 
+		find_a_world:
+		if (unknownWorld) { // we can't fetch the chunk without a world
+			for (Entity entity : entityArray) {
+				if (entity == null)
+					continue;
+				// assume it's a local teleport, use the first entity we find as a reference
+				loc = new Location(entity.getWorld(), loc.getX(), loc.getY(), loc.getZ(), loc.getYaw(), loc.getPitch());
+				break find_a_world;
+			}
+			return next; // no entities = no chunk = nobody teleporting
+		}
```

Comment:
I'm not 100% convinced this is great, but it seems like a very good assumption most of the time so whatever